src/components/containers/ConnectedMellowLpDepositForm/ConnectedMellowLpDepositForm.tsx

```diff
@@ -0,0 +1,221 @@
+import { AugmentedMellowLpVault } from "@utilities";
+import { routes } from '@routes';
+import { useDispatch } from "react-redux";
+import { useNavigate } from "react-router-dom";
+import MellowLpDepositForm from "src/components/interface/MellowLpDepositForm/MellowLpDepositForm";
+import { Panel } from "src/components/atomic";
+import React, { useState, useEffect, ReactNode } from "react";
+import { colors } from '@theme';
+import { Box } from "@mui/system";
+
+
+export type ConnectedMellowLpDepositFormProps = {
+    vault: AugmentedMellowLpVault;
+    onReset: () => void;
+}
+
+export enum DepositStates {
+    INITIALISING = "INITIALISING",
+    PROVIDER_ERROR = "PROVIDER_ERROR",
+    APPROVE_REQUIRED = "APPROVE_REQUIRED",
+    APPROVING = "APPROVING",
+    APPROVE_FAILED = "APPROVE_FAILED",
+    APPROVED = "APPROVED",
+    DEPOSITING = "DEPOSITING",
+    DEPOSIT_FAILED = "DEPOSIT_FAILED",
+    DEPOSIT_DONE = "DEPOSIT_DONE",
+};
+
+type TextProps = {
+    bold?: boolean;
+    children?: ReactNode;
+    green?: boolean;
+    red?: boolean;
+};
+const Text = ({ bold, children, green, red }: TextProps) => (
+    <Box component='span' sx={{
+        color: green ? colors.vzCustomGreen1 : red ? colors.vzCustomRed1 : undefined,
+        fontWeight: bold ? 'bold' : undefined,
+        textTransform: 'none'
+    }}>
+        {children}
+    </Box>
+);
+
+const ConnectedMellowLpDepositForm: React.FunctionComponent<ConnectedMellowLpDepositFormProps> = ({ vault, onReset }) => {
+    const navigate = useNavigate();
+
+    const [selectedDeposit, setSelectedDeposit] = useState<number>(0);
+
+    const [depositState, setDepositState] = useState<DepositStates>(DepositStates.INITIALISING);
+    const [error, setError] = useState<ReactNode>();
```

Comment:
```suggestion
    const [error, setError] = useState<string>();
```

ReactNode is super heavy object. I would not keep it in the state. Keep the string then render the string if the string is not null or empty

Reply:
Resolved. https://github.com/Voltz-Protocol/voltz-ui/pull/295/commits/a44c3c48e2a2660e1fc14f6eeeda646152d2edea